File: 624B.cpp
#include<bits/stdc++.h>
using namespace std;

int a[200], p[200];

bool ifSorted(int n, int m){

  int leftMax = 0;
  int currentMax = a[0];

  for(int i=1;i<n;i++){
    if(!p[i-1])
      leftMax = currentMax;

    if(leftMax>a[i])
      return false;

    currentMax = max(a[i],currentMax);
  }

  return true;
}

int main(){

  int t, m , n, temp;
  cin>>t;

  for(int i=0;i<t;i++){

      cin>>n>>m;
      for(int j=0;j<n;j++)
        p[j] = 0;

      for(int j=0;j<n;j++){
        cin>>a[j];
      }
      for(int j=0;j<m;j++){
        cin>>temp;
        p[temp-1]=1;
      }

     bool ret = ifSorted(n,m);
     if(ret)
      cout<<"YES"<<endl;
    else
      cout<<"NO"<<endl; 

  }

}

File: 624B.py

import numpy as np
import sys

def ifSorted(n:int, m:int, a:list, p:list):
    maxLeft = 0 #this stores the max from the left of the unswappable position
    currentMax = a[0]

    i = 1
    while i<n:

        #if the position before the current position is unswappable
        #make the current max the max from the left of the left most unswappable position
        #left most unswappable position is the left most index of p with 0 in it
        if p[i-1]==0:
            maxLeft = currentMax

        if maxLeft>a[i]:
            return False

        currentMax = max(a[i], currentMax)
        i+=1

    return True
if __name__ == "__main__":
    lines = sys.stdin.readlines()
    t = int(lines[0])

    for case in range(t):
        index = case*3 + 1

        n,m = map(int, lines[index].split())
        a = [int(x) for x in lines[index+1].split()]
        temp = [int(x) for x in lines[index+2].split()]
        p = np.zeros(n)

        for pos in temp:
            p[pos - 1] = 1
        ret = ifSorted(n, m, a, p)
        if ret:
            print("YES")
        else:
            print("NO")

    # print(a)
    # print(p)

 


